### PR TITLE
Node v4 support

### PR DESCRIPTION
**Pull request by mrollinsiv**
Removed default params.  ES6, Babel and transpiling for NPM will be added in #8

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,7 +52,8 @@ module.exports = function(parent, theirs, mine, options = {}) {
  * @param {object} options - Key specific options for diff
  * @returns {Array} - Array of objects specifying the differences
  */
-function recurse(parent, theirs, mine, path = [], options = {}) {
+function recurse(parent, theirs, mine, path, options) {
+  path = path || [];
```
Comment (shawnpwalsh) [inline index.js:56]: No `options = options || {};`?

File: index.js [hunk 2]
```diff
@@ -88,7 +91,7 @@ function recurse(parent, theirs, mine, path = [], options = {}) {
  * @param {object} options - Key specific options for diff
  * @returns {Array} - Array of objects specifying the differences
  */
-function processKeyValuePair(key, value, parent, theirs, mine, path, options = {}) {
+function processKeyValuePair(key, value, parent, theirs, mine, path, options) {
```
Comment (shawnpwalsh) [inline index.js:95]: No `options = options || {};`?

File: index.js [hunk 3]
```diff
@@ -101,7 +104,7 @@ function processKeyValuePair(key, value, parent, theirs, mine, path, options = {
       // If single dimension array run the comparison
       if (literalValues.length == value.length) {
         path.push(key);
-        var differences = compareValues(parent[key], theirs[key], mine[key], path, options[key]);
+        var differences = compareValues(parent[key], theirs[key], mine[key], path, options[key] || {});
```
Comment (shawnpwalsh) [inline index.js:109]: Think I'd rather see this handled in `compareValues` than in the caller

File: index.js [hunk 4]
```diff
@@ -143,7 +146,7 @@ function processKeyValuePair(key, value, parent, theirs, mine, path, options = {
  * @param {object} options - Key specific options for diff
  * @returns {object|null} - An object containing the conflict/edit, otherwise empty return
  */
-function compareValues(parent, theirs, mine, path, options = {}) {
+function compareValues(parent, theirs, mine, path, options) {
```
Comment (shawnpwalsh) [inline index.js:151]: No `options = options || {};`?

File: index.js (hunk 1 above)
Comment (mrollinsiv) [reply index.js:56]: It's set on line 26, and then the calls to this function provide defaults (lines 107, 120, 127)

File: index.js (hunk 2 above)
Comment (mrollinsiv) [reply index.js:95]: See above comment

File: index.js (hunk 3 above)
Comment (mrollinsiv) [reply index.js:109]: options[key] may not exist, I'll have to add the default at the start of this function.

File: index.js (hunk 4 above)
Comment (mrollinsiv) [reply index.js:151]: See above comment

File: index.js (hunk 3 above)
Comment (shawnpwalsh) [reply index.js:109]: In that case, I'm ok with this usage

File: index.js (hunk 1 above)
Comment (shawnpwalsh) [reply index.js:56]: As discussed out loud, protect against the function erroring out depending on parameters.

Comment (mrollinsiv) [reply index.js:56]: All defaults have been added

**Review by shawnpwalsh — APPROVED** [review]

**Pull request merged**